src/CodeGeneration.Roslyn.Tool/Program.cs

```diff
@@ -72,15 +74,13 @@ private static async Task<int> Core(string[] args, CancellationToken cancellatio
                 return 2;
             }
 
-            var generator = new CompilationGenerator
+            if (projectDir == null)
             {
-                ProjectDirectory = projectDir,
-                Compile = Sanitize(compile),
-                ReferencePath = Sanitize(refs),
-                PreprocessorSymbols = preprocessorSymbols,
-                PluginPaths = Sanitize(plugins),
-                IntermediateOutputDirectory = outputDirectory,
-            };
+                Console.Error.WriteLine("The project directory must be specified.");
+                return 3;
```

Comment:
Why is this changed to be required? ProjectDir is `string?` all the way down to TransformationContext.

Reply:
The commandline declaration states that `projectDir` is required. Therefore I implied, that in this case (from the generator's perspective) the project directory must not be `null`.

```CSharp
syntax.DefineOption("projectDir", ref projectDir, true, "The absolute path of the directory where the project file is located");
```
```

Reply:
Interesting. Thanks for clarification.